yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
+        if proj_dim not in [2, 3]:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.proj_dim = proj_dim
+        self.alpha = alpha
+        self._cax = None
+
+    @property
+    def cax(self):
+        """
+        The axes of the colorbar, right of the scatterplot.
+        """
+        if self._cax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._cax       
+        
+    def _layout_colorbar(self):
+        """
+        Creates the layout for colorbar. The colorbar is added to the right of the 
+        scatterplot 
+        """
+        # Ensure matplotlib version compatibility
+        if make_axes_locatable is None:
+            raise YellowbrickValueError((
+                "heatmap requires matplotlib 2.0.2 or greater "
+                "please upgrade matplotlib or set heatmap=False on the visualizer"
+            ))
+
+        # Create the new axes for the colorbar
+        divider = make_axes_locatable(self.ax)
+        self._cax = divider.append_axes("right", size="5%", pad=0.3)
+        
+    def fit(self, X, y=None, **kwargs):
+        
+        self.fit_transform(X, y, **kwargs)
+        return self
+        
+    def fit_transform(self, X, y=None, **kwargs):
+
+        # Transformer will be  defined either in the subclasses or explicitly
+        self.transformed_features_ = self.transformer.fit_transform(X)
+        super(ProjectionVisualizer, self).fit(X, y, **kwargs)
+        return self.transformed_features_    
+        
+    def transform(self, X, y=None, **kwargs):
+        try:
+            return self.transformer.transform(X)
+        except AttributeError as e:
+            raise AttributeError(str(e) + " try using fit_transform instead.")
+
+    # Not sure about the function signature      
+    def draw(self, X=None, y=None):
+        X = self.transformed_features_
+        scatter_kwargs = self._determine_scatter_kwargs(y);
+        
+        if self.proj_dim == 2:
+            self._scatter = self.ax.scatter(X[:,0], X[:,1], **scatter_kwargs)
+
+        if self.proj_dim == 3:
+            self.fig = plt.figure()
```

Comment:
Rather than creating a new figure, I'd prefer if we used `plt.gcf` (get current figure) - and also I don't think we should store the fig on the visualizer as discussed in #826.